ios/.npmignore

```diff
@@ -0,0 +1,6 @@
+*/project.xcworkspace/
+*/xcuserdata/
+.DS_Store
+.npmignore
+Pods/
```

Comment:
@friederbluemle just a question - I notice you put sub-.npmignore files in here and android and I just recently published a react-native-device-info package with accidental inclusions of stuff here, even though I attempted to npmignore them at the top level. Have you seen that before? Is that why you have layered npmignore going on in this PR?

Reply:
Hey @mikehardy - Good question. The way NPM handles npmignore, gitignore, and the files field in package.json is pretty odd and not intuitive. There are a bunch of special cases if one file exists but not the other, the `files` field is present or not, if matches are located at the root level or in a subfolder etc.
The reason I put two `.npmignore` files in the `android/` and `ios/` folder is because the very precisely ignore only files related to that platform.
Since `android/` and `ios/` are _included_ using the `files` field in `package.json`, it is _not_ possible to just use one `.npmignore` file at the root level to exclude files from those directories. Hope that makes sense.

Reply:
Clear as mud :-), but the tools' fault not yours as it is not easy to un-tangle how the different include/excludes work (what priority, whitelist or blacklist, when directives short-circuit other directives etc). For now in RNDI I just do a clean pre-package but I had not thought of nested npmignore files. Thanks for the explanation

Reply:
Here's a good article that explains some of it in more detail:
https://medium.com/@jdxcode/for-the-love-of-god-dont-use-npmignore-f93c08909d8d

I also opened this one in `react-native-device-info` with similar changes:
https://github.com/react-native-community/react-native-device-info/pull/872